packages/platform-core/src/telemetry/get_installation_id.ts

```diff
@@ -0,0 +1,20 @@
+import { userInfo } from 'os';
+import { v5 as uuidV5 } from 'uuid';
+import { CwdPackageJsonLoader } from '..';
+
+// eslint-disable-next-line spellcheck/spell-checker
+const AMPLIFY_CLI_UUID_NAMESPACE = 'e7368840-2eb6-4042-99b4-9d6c2a9370e6'; // A random v4 UUID
+/**
+ * Generates a consistent installation Uuid from the cwd + packageJsonName + userName
+ */
+export const getInstallationUuid = async (
+  namespace: string = AMPLIFY_CLI_UUID_NAMESPACE
+) => {
+  const packageJsonName = (
+    await new CwdPackageJsonLoader().loadCwdPackageJson()
+  ).name;
+  const userName = userInfo().username;
+  const cwdPath = process.cwd();
+
+  return uuidV5(cwdPath + packageJsonName + userName, namespace);
```

Comment:
should this mix some "machine identifier" like mac address ?

Reply:
the network interfaces could be named differently and some have just all 0s in it. Would this be a better idea https://www.npmjs.com/package/node-machine-id?

Reply:
The network interface was an example. That library looks promising if it's maintained. So feel free to find best solution for this. We need some machine id.

Reply:
This doesn't look well supported or a good alternative that exists without requiring a local state.

Reply:
We shouldn't include cwd in this seed. You can run `npx amplify` from anywhere in your directory. It should be the path of the root package.json or something like that

Reply:
Good point, I don't think it's possible with `npx` to figure out the "root" app since customer can have many nested workspaces. The only other reliable option I can think of is persisting a UUID along with consent information.

Reply:
How about rendering a file next to `package.json` in `node_modules` where this package gets installed ?

Reply:
One more idea.
We could hash absolute path derived from `import.meta.url` , that should be stable.

Reply:
`import.meta.url` is a good idea, but platform-core is built into `commonJs` and doesn't like `import.meta` :(

Reply:
`npx which amplify` would probably get what we want

Reply:
But that's ipc call.
how about `__dirname` ?

Reply:
Isn't `__dirname` same as `process.cwd`?

Reply:
No. it's dir of current file afaik https://nodejs.org/api/modules.html#__dirname

Reply:
I'll give it a shot

Reply:
No __dirname could work. That's the path of the actual source code file that is executing which should be stable for a given installation

Reply:
Now I'm wondering if we should _only_ seed with `__dirname`? The packageJsonName and userName could both change without actually being a different installation. `__dirname` should be stable for a given installation in all normal scenarios I can think of

Reply:
`__dirname` + some "machine identifier". (if we punt machine id we should create gh issue)

Reply:
userName changing is unlikely though it's usually is included in the path (for unix at least), however for cases where users are installing an "example" app in a /tmp or a root directory, we will get same installation ids across different users.

There is no reliable way to get machine identifier as per my quick research. We will have to punt that.

Reply:
Updated to use `__dirname` for now

Reply:
Long term - https://github.com/aws-amplify/amplify-backend/issues/665